e2e/cypress/tests/integration/performance/team_switch_spec.js

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Stage: @prod
+// Group: @channel
+import {measurePerformance} from './utils.js';
+
+describe('Channel switch performance test', () => {
+    let testUser;
+    let testTeam2;
+    before(() => {
+        cy.apiInitSetup().then(({team, user}) => {
+            testUser = user;
+
+            // # Login as test user and go to town square
+            cy.apiLogin(testUser);
+            cy.visit(`/${team.name}/channels/town-square`);
+        });
+
+        cy.apiCreateTeam('team-b', 'Team B').then(({team}) => {
+            testTeam2 = team;
+        });
+    });
+
+    it('measures switching between two teams from LHS', () => {
+        // # Invoke window object
+        measurePerformance('teamLoad', 5000, () => {
+            // # Switch to Team 2
+            cy.get('#teamSidebarWrapper').within(() => {
+                cy.get(`#${testTeam2.name}TeamButton`).click();
+            });
+
+            // * Expect that the user has switched teams
+            expectActiveTeamToBe(testTeam2.display_name, testTeam2.name);
+        });
```

Comment:
IMO, the measurement here becomes "tool-dependent" (or in this case, depends to Cypress). I honestly favored the approach being taken in the [mobile app](https://github.com/mattermost/mattermost-mobile/pull/6700/files) where the instrumentation is placed inside the app and makes it "tool-agnostic".

Reply:
@saturninoabril Being tool dependant was kind of "desired" (at least in the first approach). Not having enough granularity is what made the mobile approach to add the instrumentation in the code.

And why would we not want to add the instrumentation on the code? Mainly because it will be harder to create new performance tests, and it will add noise to the code. If we can keep all the instrumentation out of the code, and move it to the tool, I think will make things easier (until we change the tool, of course, then it will be really annoying :sweat_smile:)